src/main/java/edu/harvard/dbmi/avillach/dictionaryetl/loading/DictionaryLoaderService.java

```diff
@@ -115,83 +136,137 @@ public String processColumnMetaCSV(String csvPath, String errorFile) throws Runt
-     *                    variables can have multiple rows in the columnMeta.csv.
-     */
     protected void processColumnMetas(List<ColumnMeta> columnMetas) {
         ColumnMeta columnMeta;
-        ConceptNode rootConceptNode;
-        try {
-            if (columnMetas.size() == 1) {
-                columnMeta = columnMetas.getFirst();
+        if (columnMetas.size() == 1) {
+            columnMeta = columnMetas.getFirst();
+        } else {
+            boolean isContinuous = columnMetas.stream().anyMatch(meta -> !meta.categorical());
+            if (!isContinuous) {
+                columnMeta = flattenCategoricalColumnMeta(columnMetas);
             } else {
-                boolean isContinuous = columnMetas.stream().anyMatch(meta -> !meta.categorical());
-                if (!isContinuous) {
-                    columnMeta = flattenCategoricalColumnMeta(columnMetas);
-                } else {
-                    columnMeta = flattenContinuousColumnMeta(columnMetas);
-                }
+                columnMeta = flattenContinuousColumnMeta(columnMetas);
             }
-
-            rootConceptNode = buildConceptHierarchy(columnMeta.name());
-            createDatabaseEntries(rootConceptNode, columnMeta);
-        } catch (Exception e) {
-            log.error("Error processing concept path: {} with values for column metas: {}",
-                    columnMetas.getFirst().name(),
-                    e.getMessage());
-            columnMetaErrors.add(columnMetas);
-        } finally {
-            // Decrement the task counter no matter what happens
-            task.getAndDecrement();
         }
+
+        this.concurrentFullPathTree.ingestColumnMeta(columnMeta);
     }
 
-    /**
-     * In some cases we need to flatten a group of ColumnMetaRows down to a single ColumnMeta.
-     * This is the case for Continuous Concept Paths in the columnMeta.csv that have more than one row.
-     *
-     * @param columnMetas A List of ColumnMeta where the first ColumnMeta is expected to be a continuous value.
-     * @return ColumnMeta that has a min and max based on all ColumnMetas in the list.
-     */
     private ColumnMeta flattenContinuousColumnMeta(List<ColumnMeta> columnMetas) {
-        // This is a special case. Where the parent (first element) being rolled into must be continuous.
         if (columnMetas.getFirst().categorical()) {
             throw new RuntimeException("Cannot flatten rows. Mixed concept types. Must be continuous OR categorical " +
                                        "for a concept path.");
         }
 
-        // As the list is processed the min and max will adjust to based on the "values" of other concepts.
         final Double[] min = {columnMetas.getFirst().min()};
-        final Double[] max = {columnMetas.getLast().max()};
+        final Double[] max = {columnMetas.getFirst().max()};
```

Comment:
Why am I doing this? Investigate. If the columnMetas is length of one .getFirst() and .getLast() should return the same element. Why do I need to do this. 🤔